### DISCUSSION TITLE
QrBill without amount

### DISCUSSION THREAD

**Discussion by Hucky69** [manuelbl/SwissQRBill.NET#56] [Ideas]
Hi your tool is nearly perfect. What I missing is the possibility to create a SwissQrBill without a amount.
Some customers need a bill without amount. In this case the SwissQrBill shout create 4 corners so that the debtor can write a amount. This should be easy if the amount = zero then create this 4 corners.
Is it possible to develope this? This would be great.

**Comment by manuelbl** [comment]
That already exists:

```
// Setup bill data
Bill bill = new Bill
{
    // creditor data
    Account = "CH4431999123000889012",
    Creditor = ...,

    // payment data
    Amount = null,
    ...
};
```

![QR bill without amount](https://user-images.githubusercontent.com/10954524/183582010-992b0710-3070-4d45-8161-6a17d70f1e33.svg)

Note that *Amount = 0* is something different. It's part of a formal invoice for an order that has already been paid (e.g. by credit card). But for technical reasons, the invoice still contains the QR bill payment slip. The text "DO NOT USE FOR PAYMENT" will not be automatically shown. It must be explicitly set in the bill data.

![QR bill not for payment](https://user-images.githubusercontent.com/10954524/183582172-a954c205-c4d1-4d92-b0b5-d6ab85a9315f.svg)

**Comment by Hucky69** [comment]
Hallo Manuel, vielen Dank für die rasche Antwort. Ich werde das ausprobieren

 

 

Von: Manuel Bl. <> 
Gesendet: Dienstag, 9. August 2022 08:46
An: manuelbl/SwissQRBill.NET ***@***.***>
Cc: Hucky69 ***@***.***>; Author ***@***.***>
Betreff: Re: [manuelbl/SwissQRBill.NET] QrBill without amount (Discussion #56)

 

That already exists:

// Setup bill data
Bill bill = new Bill
{
    // creditor data
    Account = "CH4431999123000889012",
    Creditor = ...,
 
    // payment data
    Amount = null,
    ...
};

 <https://user-images.githubusercontent.com/10954524/183582010-992b0710-3070-4d45-8161-6a17d70f1e33.svg> 

Note that Amount = 0 is something different. It's part of a formal invoice for an order that has already been paid (e.g. by credit card). But for technical reasons, the invoice still contains the QR bill payment slip. The text "DO NOT USE FOR PAYMENT" will not be automatically shown. It must be explicitly set in the bill data.

 <https://user-images.githubusercontent.com/10954524/183582172-a954c205-c4d1-4d92-b0b5-d6ab85a9315f.svg>
